code/modules/client/client procs.dm

```diff
@@ -304,7 +304,7 @@
 	if(byond_version < SUGGESTED_CLIENT_VERSION) // Update is suggested, but not required.
 		to_chat(src,"<span class='userdanger'>Your BYOND client (v: [byond_version]) is out of date. This can cause glitches. We highly suggest you download the latest client from http://www.byond.com/ before playing. </span>")
 
-	if(IsGuestKey(key))
+	if(IsGuestKey(key) && !GLOB.guests_allowed)
 		alert(src,"This server doesn't allow guest accounts to play. Please go to http://www.byond.com/ and register for a key.","Guest","OK")
 		qdel(src)
 		return
```

Comment:
Remove this entire proc. We already check if a user is a guest in `world/IsBanned()`, and the `alert()` here wont even show up, since we delete their client the next line.

Reply:
done